crates/bitwarden-auth/src/identity/login_via_password/login_via_password_impl.rs

```diff
@@ -0,0 +1,592 @@
+use bitwarden_core::key_management::MasterPasswordAuthenticationData;
+#[cfg(feature = "wasm")]
+use wasm_bindgen::prelude::*;
+
+use crate::identity::{
+    LoginClient,
+    api::{request::LoginApiRequest, send_login_request},
+    login_via_password::{PasswordLoginApiRequest, PasswordLoginError, PasswordLoginRequest},
+    models::LoginResponse,
+};
+
+#[cfg_attr(feature = "wasm", wasm_bindgen)]
+#[cfg_attr(feature = "uniffi", uniffi::export)]
+impl LoginClient {
+    /// Authenticates a user via email and master password.
+    ///
+    /// Derives the master password hash using KDF settings from prelogin, then sends
+    /// the authentication request to obtain access tokens and vault keys.
+    ///
+    /// # Errors
+    ///
+    /// - [`PasswordLoginError::InvalidUsernameOrPassword`] - Invalid credentials
```

Comment:
Style: Generally errors are not documented like this, and you are better off just going to definition of the error in rust docs / your IDE and reading through the cases.